core/src/cluster_nodes.rs

```diff
@@ -0,0 +1,441 @@
+use {
+    crate::{broadcast_stage::BroadcastStage, retransmit_stage::RetransmitStage},
+    itertools::Itertools,
+    solana_gossip::{
+        cluster_info::{compute_retransmit_peers, ClusterInfo},
+        contact_info::ContactInfo,
+        crds_gossip_pull::CRDS_GOSSIP_PULL_CRDS_TIMEOUT_MS,
+        weighted_shuffle::{weighted_best, weighted_shuffle},
+    },
+    solana_sdk::pubkey::Pubkey,
+    std::{any::TypeId, cmp::Reverse, collections::HashMap, marker::PhantomData},
+};
+
+enum NodeId {
```

Comment:
How about `NodeGossipStatus`, with the variants being, `AliveNode(ContactInfo)`, and `DeadStakedNode(Pubkey)`

Reply:
This struct is meant to include node identifying info. No need to tie it to gossip. Also, this is only a private type not exposed outside of this file, so leaning on brevity here.

Reply:
fine by me!